linux/system/repo.sls

```diff
@@ -91,6 +91,16 @@ linux_repo_{{ name }}_key:
 
 {%- elif repo.key_url|default(False) %}
 
+{%- if system.proxy.get('pkg', {}).get('enabled', False) and system.proxy.get('pkg', {}).get('http', None) %}
+
+linux_repo_{{ name }}_key:
+  cmd.wait:
+    - name: "curl --proxy {{ system.proxy.get('pkg', {}).get('http', None) }}  -s {{ repo.key_url }} | apt-key add -"
```

Comment:
better to start using: https://github.com/salt-formulas/salt-formula-linux/blame/master/linux/system/repo.sls#L84

provide key on pillar. That will be more secure as well as proxy tolerant.